incubator/binding-grpc/src/main/java/io/aklivity/zilla/runtime/binding/grpc/internal/stream/GrpcServerFactory.java

```diff
@@ -0,0 +1,1369 @@
+            GrpcServer server,
+            long traceId,
+            long authorization);
+    }
+
+    public GrpcServerFactory(
+        GrpcConfiguration config,
+        EngineContext context)
+    {
+        this.writeBuffer = context.writeBuffer();
+        this.bufferPool = context.bufferPool();
+        this.signaler = context.signaler();
+        this.streamFactory = context.streamFactory();
+        this.supplyInitialId = context::supplyInitialId;
+        this.supplyReplyId = context::supplyReplyId;
+        this.supplyTraceId = context::supplyTraceId;
+        this.bindings = new Long2ObjectHashMap<>();
+        this.grpcTypeId = context.supplyTypeId(GrpcBinding.NAME);
+        this.httpTypeId = context.supplyTypeId(HTTP_TYPE_NAME);
+    }
+
+    @Override
+    public void attach(
+        BindingConfig binding)
+    {
+        GrpcBindingConfig grpcBinding = new GrpcBindingConfig(binding);
+        bindings.put(binding.id, grpcBinding);
+    }
+
+    @Override
+    public void detach(
+        long bindingId)
+    {
+        bindings.remove(bindingId);
+    }
+
+    @Override
+    public MessageConsumer newStream(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length,
+        MessageConsumer network)
+    {
+        final BeginFW begin = beginRO.wrap(buffer, index, index + length);
+        final long routeId = begin.routeId();
+        final long traceId = begin.traceId();
+        final long authorization = begin.authorization();
+        final long affinity = begin.affinity();
+        final long initialId = begin.streamId();
+        final long sequence = begin.sequence();
+        final long acknowledge = begin.acknowledge();
+        final OctetsFW extension = begin.extension();
+        final HttpBeginExFW httpBeginEx = extension.get(httpBeginExRO::tryWrap);
+
+        MessageConsumer newStream = null;
+
+        if (!isGrpcRequestMethod(httpBeginEx))
+        {
+            doHttpResponse(network, traceId, authorization, affinity, routeId, initialId, sequence, acknowledge,
+                HEADER_VALUE_STATUS_405, HEADER_VALUE_GRPC_INTERNAL_ERROR);
+            newStream = (t, b, i, l) -> {};
+        }
+        else
+        {
+            final GrpcBindingConfig binding = bindings.get(routeId);
+
+            final GrpcMethodConfig methodConfig = binding.resolveGrpcMethodConfig(httpBeginEx);
+
+            if (methodConfig != null)
+            {
+                newStream = newInitialGrpcStream(begin, network, httpBeginEx, methodConfig);
+            }
+            else
+            {
+                doHttpResponse(network, traceId, authorization, affinity, routeId, initialId, sequence, acknowledge,
+                    HEADER_VALUE_STATUS_200, HEADER_VALUE_GRPC_UNIMPLEMENTED);
+                newStream = (t, b, i, l) -> {};
+            }
+        }
+
+        return newStream;
+    }
+
+    public MessageConsumer  newInitialGrpcStream(
```

Comment:
Please make method `private` and remove double space between return type and method name.